Userland/Libraries/LibJS/Runtime/PropertyName.cpp

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PropertyName.h"
+#include "AbstractOperations.h"
+#include "GlobalObject.h"
+
+namespace JS {
+
+Value PropertyName::to_canonical_numeric_index_string(GlobalObject& global_object) const
+{
+    VERIFY(is_number() || is_string());
+
+    if (is_number())
+        return Value(m_number);
+
+    return canonical_numeric_index_string(global_object, js_string(global_object.vm(), m_string));
+}
```

Comment:
Hmm, not sure I like this wrapper - looking at the spec, `CanonicalNumericIndexString` is only ever used with a property key argument, so can we maybe update `canonical_numeric_index_string()` to take a `PropertyName const&` and reuse that instead?

It can still include the "is number? return early" optimization

Reply:
Yes that was the other option I considered.
Do we still want a `canonical_numeric_index_string()` overload for Value?

Reply:
Since that would currently be unused, no - we can add it once necessary :)

Reply:
Since there is currently no use for Value I didn't add the overload